cloud/docker/docker.py

```
@@ -657,6 +671,12 @@ def start_containers(self, containers):
             params['dns'] = self.module.params.get('dns')
             params['volumes_from'] = self.module.params.get('volumes_from')
 
+        if docker.utils.compare_version('1.14', self.client.version()['ApiVersion']) >= 0 and hasattr(docker, '__version__') and docker.__version__ >= '0.5.0':
+            if self.module.params.get('restart_policy') is not None:
+                params['restart_policy'] = { 'Name': self.module.params.get('restart_policy') }
+                if params['restart_policy']['Name'] == 'on-failure':
+                    params['restart_policy']['MaximumRetryCount'] = self.module.params.get('restart_policy_retry')
+
         for i in containers:
```

It looks like this will silently ignore the restart_policy parameter if one of the docker-related versions is too old.  Is that correct?  Probably better to first check if restart_policy was specified.  If so, then return an error if the docker version is too old.